### PR TITLE
Add profiler skip filter

### PR DESCRIPTION
**Pull request by sam-at-luther**

### CODE REVIEW

File: lisp/x/profiler/callgrind.go [hunk 1]
```diff
@@ -183,67 +172,41 @@ func (p *callgrindProfiler) getCallRefAndDecrement() *callRef {
...
-			fmt.Fprintf(p.writer, "cfl=%s\n", p.getRef(entry.file))
-			fmt.Fprintf(p.writer, "cfn=%s\n", p.getRef(entry.name))
-			fmt.Fprint(p.writer, "calls=1 0 0\n")
-			fmt.Fprintf(p.writer, "%d %d %d\n", entry.line, entry.duration, memory)
-		}
-		// and end the entry
-		fmt.Fprint(p.writer, "\n")
-	default:
-		panic(fmt.Sprintf("Missing type %d", function.Type))
+	fName := prettyFunName(p.runtime, fun)
```
Comment (jeremy-asher) [inline lisp/x/profiler/callgrind.go:181]: Is this safe to call all the time? Looks like it can panic.

File: lisp/x/profiler/go_annotator_test.go [hunk 2]
```diff
@@ -23,79 +23,12 @@ func TestNewPprofAnnotator(t *testing.T) {
...
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-		9 7 8 6 5 4 3 2 8 36 73 83 3 82 2 1 334 3 82 1 32 3 8 0 1 2 83 4 9 0 11 3 4
-	)
-)
-`)
+	testsrc := env.LoadString("test.lisp", longTestLisp)
```
Comment (jeremy-asher) [inline lisp/x/profiler/go_annotator_test.go]: Should this be `longtest.lisp`?

File: lisp/x/profiler/profiler.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+// skipTrace is a helper function to decide whether to skip tracing.
+func (p *profiler) skipTrace(v *lisp.LVal) bool {
+	return !p.enabled || defaultSkipFilter(v) || p.skipFilter != nil && p.skipFilter(v)
+}
+
+var builtinRegex = regexp.MustCompile("\\<(?:builtin|special)-[a-z]+ \\`\\`(.*)\\'\\'\\>")
+
+// Gets a canonical version of the function name suitable for viewing in KCacheGrind
```
Comment (jeremy-asher) [inline lisp/x/profiler/profiler.go]: Should this comment be generalized?

File: lisp/x/profiler/callgrind.go (hunk 1 above)
Comment (sam-at-luther) [reply lisp/x/profiler/callgrind.go:181]: Can it?I think I added a guard around it to avoid the panics--did I miss one?

Comment (jeremy-asher) [reply lisp/x/profiler/callgrind.go:181]: `prettyFunName` immediately calls `fun.FunData` which can panic depending on the type I believe

Comment (sam-at-luther) [reply lisp/x/profiler/callgrind.go:181]: Good call. Updated.

**Review by jeremy-asher — APPROVED** [review]

**Pull request merged**